android/app/src/main/java/me/kisoft/covid19/HealthWatcher/VitalSignsProcess.java

```diff
@@ -77,11 +79,46 @@
     public ArrayList<Double> BlueAvgList = new ArrayList<>();
     public int counter = 0;
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+            case CAMERA_PERMISSION_REQUEST_CODE: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
+                    // camera-related task you need to do.
+
+                } else {
+                    // permission denied, boo! Disable the
+                    // functionality that depends on this permission.
+                    previewHolder.addCallback(surfaceCallback);
+                    previewHolder.setType(SurfaceHolder.SURFACE_TYPE_PUSH_BUFFERS);
+                    camera = Camera.open();
+                    camera.setDisplayOrientation(90);
+
+                }
+                return;
+            }
+
+            // other 'case' lines to check for other
+            // permissions this app might request.
+        }
+    }
+
```

Comment:
I Tested the permission on my phone and this works !

Reply:
But, When I deleted the app and tried it, the request ws made too late so it crahsed. We need to do the request before. Maybe at the beggining of the app ?